tensorflow/contrib/distribute/python/parameter_server_strategy.py

```diff
@@ -296,7 +296,7 @@ def _verify_destinations_not_different_worker(self, destinations):
       return
     for d in cross_tower_ops_lib.get_devices_from(destinations):
       d_spec = tf_device.DeviceSpec.from_string(d)
-      if d_spec.job == self._task_type and d_spec.task != self._task_id:
+      if d_spec.job == self._task_type and d_spec.task != (self._task_id or 0):
```

Comment:
I don't understand the expression  d_spec.task != (self._task_id or 0). What does that mean? Can you rewrite it?

Reply:
When running locally through `tf.estimator.train_and_evaluate`, the destination is set to `"/device:CPU:0"`. The `d_spec` obtained would have `d_spec.job == None` and `d_spec.task == 0`.

However, `ParameterServerStrategy` assigns both `self._task_type` and `self._task_id` to `None` for local execution, and this line always evaluates to `True` as the first expression becomes `None == None` and the second becomes `0 != None`, which results in an error thrown and effectively disallows local usage.

There are several changes I can think of to tackle this problem, one would be to set the initial value for `self._task_id` as `0` in `_initialize_local`, or change this line to consider `None` as `0` in the check. The former approach does solve this problem, but I'm not if that would break other code that depends on `self._task_id` being initially `None`, I went for the second approach to minimize the impact of the change.

P.S. The expression is equivalent to the following

```python
if not self._task_id:
  task_id = 0
else:
  task_id = self._task_id
if d_spec.job == self._task_type and d_spec.task != task_id:
  ...
```

Reply:
Yeah, I just think this particular Pythonic use or "or" is fairly unreadable. Perhaps write it as you did above or similar?

Reply:
@rmlarsen Changed the expression to a equivalent that makes use of a local variable. Please review